frontend/src/pages/Orders.js

```diff
@@ -0,0 +1,126 @@
+import { useState, useEffect, useMemo } from "react";
+import { useTable } from "react-table";
+import Axios from "axios";
+import { Container, Row, Col } from "react-bootstrap";
+
+const Orders = () => {
+  const [listOfOrders, setListOfOrders] = useState([]);
+
+  useEffect(() => {
+    Axios.get(`${process.env.REACT_APP_BASEURL}/getOrders`).then((res) => {
+      setListOfOrders(res.data);
+    });
+  }, []);
+
+  //table columns for react-table
+  const columns = useMemo(
+    () => [
+      {
+        Header: "Column 1",
+        accessor: "col1", // accessor is the "key" in the data
+      },
+      {
+        Header: "Column 2",
+        accessor: "col2",
+      },
+    ],
+    []
+  );
+  const data = useMemo(
+    () => [
+      {
+        col1: "Hello",
+        col2: "World",
+      },
+      {
+        col1: "react-table",
+        col2: "rocks",
+      },
+      {
+        col1: "whatever",
+        col2: "you want",
+      },
+    ],
+    []
+  );
+
+  const tableInstance = useTable({ columns, data });
+
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
+    tableInstance;
+
+  return (
+    // <Container className="page-size">
+    //   {listOfOrders.map((order) => (
+    //     <div key={order._id}>
+    //       {order.pizzalist.map((pizza) => (
+    //         <div key={pizza.pizzaid}>
+    //           {pizza.pizzaid}
+    //           {pizza.category}
+    //           {pizza.pizzatype}
+    //           {pizza.qty}
+    //         </div>
+    //       ))}
+    //       <div>Total ${order.total}</div>
+    //     </div>
+    //   ))}
+    // </Container>
+
+    // apply the table props
+    <table {...getTableProps()}>
+      <thead>
+        {
+          // Loop over the header rows
+          headerGroups.map((headerGroup) => (
+            // Apply the header row props
+            <tr {...headerGroup.getHeaderGroupProps()}>
+              {
+                // Loop over the headers in each row
+                headerGroup.headers.map((column) => (
+                  // Apply the header cell props
+                  <th {...column.getHeaderProps()}>
+                    {
+                      // Render the header
+                      column.render("Header")
+                    }
+                  </th>
+                ))
+              }
+            </tr>
+          ))
+        }
+      </thead>
+      {/* Apply the table body props */}
+      <tbody {...getTableBodyProps()}>
```

Comment:
This is very very complex to read and potentially add alot of complexity to when you want to go off the paht for any reason

Reply:
It might be more helpful to split up in components

Reply:
Thanks I'll check it out. Yes it definitely did add to the complexity.